contrib/flo-scio_2.12/src/main/scala/com/spotify/flo/contrib/scio/ScioOperator.scala

```diff
@@ -143,10 +147,28 @@ class ScioOperator[T] extends TaskOperator[ScioJobSpec.Provider[T], ScioJobSpec[
     spec.success(result.get)
   }
 
-  private def reportDataflowJobId(taskId: TaskId, jobId: String, listener: TaskOperator.Listener) {
-    log.info("Started scio job (dataflow): {}", jobId)
-    listener.meta(taskId, "dataflow-job-id", jobId);
+  private def reportDataflowJob(taskId: TaskId, job: DataflowPipelineJob, listener: TaskOperator.Listener) {
+    val url = dataflowJobMonitoringPageURL(job)
+    val jobMeta = Map(
+      "job-id" -> job.getJobId,
+      "project-id" -> job.getProjectId,
+      "region" -> job.getRegion,
+      "monitoring-page-url" -> url
+    )
+    log.info("Started scio job (dataflow): {}", jobMeta)
+    val jobMetaJson = new ObjectMapper().writeValueAsString(jobMeta)
+    listener.meta(taskId, "dataflow-job", jobMetaJson)
```

Comment:
Went for reporting all of these values in a single atomic piece of metadata as that seems less fragile than sending them as separate metadata values and separate requests to lineage etc.

This could be a little nicer if `listener.meta` would take a `Map<String, String>` instead.

Reply:
Let's add that API instead of storing a JSON as string.

Reply:
annoying function interface thing... maybe we break the compatibility?